internal/database/repos.go

```diff
@@ -1519,31 +1522,47 @@ func parsePattern(p string) ([]*sqlf.Query, error) {
 	return []*sqlf.Query{sqlf.Sprintf("(%s)", sqlf.Join(conds, "OR"))}, nil
 }
 
-// parseCursorConds checks whether the query is using cursor-based pagination, and
-// if so performs the necessary transformations for it to be successful.
-func parseCursorConds(c *Cursor) (conds []*sqlf.Query, err error) {
-	if c == nil || c.Column == "" || c.Value == "" {
-		return nil, nil
-	}
-	var direction string
-	switch c.Direction {
-	case "next":
-		direction = ">="
-	case "prev":
-		direction = "<="
-	default:
-		return nil, errors.Errorf("missing or invalid cursor direction: %q", c.Direction)
+// parseCursorConds returns the WHERE conditions for the given cursor
+func parseCursorConds(cs types.MultiCursor) (cond *sqlf.Query, err error) {
+	var (
+		direction string
+		operator  string
+		columns   = make([]string, 0, len(cs))
+		values    = make([]*sqlf.Query, 0, len(cs))
+	)
+
+	for _, c := range cs {
+		if c == nil || c.Column == "" || c.Value == "" {
+			continue
+		}
+
+		if direction == "" {
+			switch direction = c.Direction; direction {
+			case "next":
+				operator = ">="
+			case "prev":
+				operator = "<="
+			default:
+				return nil, errors.Errorf("missing or invalid cursor direction: %q", c.Direction)
+			}
+		} else if direction != c.Direction {
+			return nil, errors.Errorf("multi-cursors must have the same direction")
+		}
+
+		switch RepoListColumn(c.Column) {
+		case RepoListName, RepoListStars, RepoListCreatedAt, RepoListID:
+			columns = append(columns, c.Column)
+			values = append(values, sqlf.Sprintf("%s", c.Value))
+		default:
+			return nil, errors.Errorf("missing or invalid cursor: %q %q", c.Column, c.Value)
+		}
 	}
 
-	switch c.Column {
-	case string(RepoListName):
-		conds = append(conds, sqlf.Sprintf("name "+direction+" %s", c.Value))
-	case string(RepoListCreatedAt):
-		conds = append(conds, sqlf.Sprintf("created_at "+direction+" %s", c.Value))
-	default:
-		return nil, errors.Errorf("missing or invalid cursor: %q %q", c.Column, c.Value)
+	if len(columns) == 0 {
+		return nil, nil
 	}
-	return conds, nil
+
+	return sqlf.Sprintf(fmt.Sprintf("(%s) %s (%%s)", strings.Join(columns, ", "), operator), sqlf.Join(values, ", ")), nil
```

Comment:
I think this should work if you change `columns` to `[]*sqlf.Query` and operator to `*sqlf.Query`:

```suggestion
	return sqlf.Sprintf("(%s) %s (%s)", sqlf.Join(columns, ", "), operator, sqlf.Join(values, ", ")), nil
```

Reply:
That would make the columns use up bind parameters which isn't needed and makes debugging the query harder.

Reply:
you can do what you do here where the format string is the value. So

```
columns = append(columns, sqlf.Sprintf(column))
```